uint256.go

```diff
@@ -1280,34 +1280,53 @@ func (z *Int) ExtendSign(x, byteNum *Int) *Int {
 // Sqrt sets z to ⌊√x⌋, the largest integer such that z² ≤ x, and returns z.
 func (z *Int) Sqrt(x *Int) *Int {
 	// This implementation of Sqrt is based on big.Int (see math/big/nat.go).
-	if x.LtUint64(2) {
-		return z.Set(x)
+	if x.IsUint64() {
+		var (
+			x0 uint64 = x.Uint64()
+			z1 uint64 = 1 << ((bits.Len64(x0)+1)/2)
+			z2 uint64
+		)
+		if x0 < 2 {
+			return z.SetUint64(x0)
+		}
+		for {
+			z2 = (z1 + x0 / z1) >> 1
+			if z2 >= z1 {
+				return z.SetUint64(z1)
+			}
+			z1 = z2
+		}
 	}
-	var (
-		z1 = &Int{1, 0, 0, 0}
-		z2 = &Int{}
-	)
+
+	flag := false
```

Comment:
a flag called `flag` is not great for code clarity. Please rename to something expressing what it means, e.g. `first` or `notFirst`.